### ISSUE TITLE
Getting the wrong bleu when Reproducing Backtranslation Reverse Model

### ISSUE THREAD

**Issue by LiangQiqi677** [pytorch/fairseq#3381]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
I follow the backtranslation code and want to reproduce it. [https://github.com/pytorch/fairseq/tree/master/examples/backtranslation](url)
I started from **Step 1. Prepare parallel data**, and then skip to **Step 2. Back-translate monolingual German data** to train a de-en model. 
When I test at WMT17, my score is 15.+, but what you write is `# BLEU+case.mixed+lang.de-en+numrefs.1+smooth.exp+test.wmt17+tok.13a+version.1.4.3 = 34.9 66.9/41.8/28.5/19.9 (BP = 0.983 ratio = 0.984 hyp_len = 63342 ref_len = 64399)`. It's a big difference between 34.9 and 15.+. 
#### Code
`fairseq_dir=/apdcephfs/share_47076/qiqiliang/pruned/code/fairseq
code_dir=/apdcephfs/share_47076/qiqiliang/pruned/train_ende_BT
data_path=$code_dir/fair_deen
CHECKPOINT_DIR=$code_dir/models_deen_baseline
export PYTHONPATH=$fairseq_dir:$PYTHONPATH
export CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7

python3 $fairseq_dir/fairseq_cli/train.py ${data_path}  \
  --fp16 \
  --source-lang de --target-lang en \
  --arch transformer_wmt_en_de_big --share-all-embeddings \
  --dropout 0.3 --weight-decay 0.0 \
  --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
  --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
  --lr 0.001 --lr-scheduler inverse_sqrt --warmup-updates 4000 \
  --max-tokens 5500 --update-freq 16 \
  --max-update 30000 \
  --save-dir $CHECKPOINT_DIR \
  --no-epoch-checkpoints \
  --save-interval-updates 1000 \
  --skip-invalid-size-inputs-valid-test \
  --num-workers 20 \`
   

#### What's your environment?

 - fairseq Version :0.9.0
 - PyTorch Version :1.14.0
 - Python version :3.6
 - CUDA/cuDNN version :10.1
 - GPU models and configuration :Tesla-V100

**Comment by LiangQiqi677** [comment]
This is the first 2 epoch:

2021-03-20 01:01:00 | INFO | train_inner | epoch 001:    105 / 941 loss=12.982, nll_loss=12.642, ppl=6389.8, wps=62073.7, ups=0.39, wpb=157560, bsz=5521.3, num_updates=100, lr=2.5e-05, gnorm=2.371, loss_scale=4, train_wall=256, wall=1026
2021-03-20 01:05:14 | INFO | train_inner | epoch 001:    205 / 941 loss=10.932, nll_loss=10.282, ppl=1245.36, wps=61787.6, ups=0.39, wpb=156802, bsz=5522.6, num_updates=200, lr=5e-05, gnorm=1.719, loss_scale=4, train_wall=252, wall=1280
2021-03-20 01:09:28 | INFO | train_inner | epoch 001:    305 / 941 loss=10.225, nll_loss=9.432, ppl=690.93, wps=61870.3, ups=0.39, wpb=157046, bsz=5531, num_updates=300, lr=7.5e-05, gnorm=1.605, loss_scale=4, train_wall=252, wall=1534
2021-03-20 01:11:09 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 01:11:17 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 01:13:42 | INFO | train_inner | epoch 001:    405 / 941 loss=9.567, nll_loss=8.673, ppl=408.07, wps=61631.1, ups=0.39, wpb=156741, bsz=5584.1, num_updates=400, lr=0.0001, gnorm=1.745, loss_scale=4, train_wall=253, wall=1788
2021-03-20 01:17:56 | INFO | train_inner | epoch 001:    505 / 941 loss=9.028, nll_loss=8.046, ppl=264.25, wps=62027.2, ups=0.39, wpb=157442, bsz=5503.6, num_updates=500, lr=0.000125, gnorm=1.607, loss_scale=4, train_wall=252, wall=2042
2021-03-20 01:22:10 | INFO | train_inner | epoch 001:    605 / 941 loss=8.623, nll_loss=7.579, ppl=191.18, wps=61769.2, ups=0.39, wpb=157038, bsz=5542.3, num_updates=600, lr=0.00015, gnorm=1.498, loss_scale=8, train_wall=253, wall=2296
2021-03-20 01:26:12 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 01:26:19 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 01:26:25 | INFO | train_inner | epoch 001:    705 / 941 loss=8.216, nll_loss=7.11, ppl=138.1, wps=61819.7, ups=0.39, wpb=157155, bsz=5571.4, num_updates=700, lr=0.000175, gnorm=1.399, loss_scale=8, train_wall=253, wall=2550
2021-03-20 01:29:50 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 4.0
2021-03-20 01:30:41 | INFO | train_inner | epoch 001:    806 / 941 loss=7.796, nll_loss=6.624, ppl=98.61, wps=61516.5, ups=0.39, wpb=157750, bsz=5511.5, num_updates=800, lr=0.0002, gnorm=1.333, loss_scale=7, train_wall=255, wall=2807
2021-03-20 01:34:54 | INFO | train_inner | epoch 001:    906 / 941 loss=7.424, nll_loss=6.194, ppl=73.2, wps=62159.5, ups=0.4, wpb=157217, bsz=5398, num_updates=900, lr=0.000225, gnorm=1.174, loss_scale=4, train_wall=252, wall=3060
2021-03-20 01:36:22 | INFO | fairseq_cli.train | begin validation on "valid" subset
2021-03-20 01:37:25 | INFO | valid | epoch 001 | valid on 'valid' subset | loss 6.817 | nll_loss 5.368 | ppl 41.29 | wps 150549 | wpb 8996.7 | bsz 315.6 | num_updates 935
2021-03-20 01:37:25 | INFO | fairseq_cli.train | begin save checkpoint
2021-03-20 01:37:38 | INFO | fairseq.checkpoint_utils | saved checkpoint /apdcephfs/share_47076/qiqiliang/pruned/train_ende_BT/models_deen_baseline/checkpoint_best.pt (epoch 1 @ 935 updates, score 6.817) (writing took 13.510601406916976 seconds)
2021-03-20 01:37:38 | INFO | fairseq_cli.train | end of epoch 1 (average epoch stats below)
2021-03-20 01:37:38 | INFO | train | epoch 001 | loss 9.338 | nll_loss 8.412 | ppl 340.57 | wps 59935.2 | ups 0.38 | wpb 157175 | bsz 5513.1 | num_updates 935 | lr 0.00023375 | gnorm 1.59 | loss_scale 5 | train_wall 2394 | wall 3224
2021-03-20 01:37:38 | INFO | fairseq_cli.train | begin training epoch 1
2021-03-20 01:48:44 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 01:49:10 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 01:51:53 | INFO | train_inner | epoch 002:     65 / 941 loss=7.048, nll_loss=5.76, ppl=54.21, wps=15398.4, ups=0.1, wpb=156897, bsz=5440.5, num_updates=1000, lr=0.00025, gnorm=1.18, loss_scale=4, train_wall=250, wall=4079
2021-03-20 01:51:53 | INFO | fairseq_cli.train | begin validation on "valid" subset
2021-03-20 01:52:56 | INFO | valid | epoch 002 | valid on 'valid' subset | loss 6.629 | nll_loss 5.143 | ppl 35.34 | wps 152053 | wpb 8996.7 | bsz 315.6 | num_updates 1000 | best_loss 6.629
2021-03-20 01:52:56 | INFO | fairseq_cli.train | begin save checkpoint
2021-03-20 01:53:17 | INFO | fairseq.checkpoint_utils | saved checkpoint /apdcephfs/share_47076/qiqiliang/pruned/train_ende_BT/models_deen_baseline/checkpoint_1000.pt (epoch 2 @ 1000 updates, score 6.629) (writing took 20.699145946651697 seconds)
2021-03-20 01:57:30 | INFO | train_inner | epoch 002:    165 / 941 loss=6.661, nll_loss=5.314, ppl=39.77, wps=46598.7, ups=0.3, wpb=157002, bsz=5558, num_updates=1100, lr=0.000275, gnorm=1.021, loss_scale=4, train_wall=252, wall=4415
2021-03-20 02:01:43 | INFO | train_inner | epoch 002:    265 / 941 loss=6.284, nll_loss=4.88, ppl=29.44, wps=62056.5, ups=0.39, wpb=157412, bsz=5441.4, num_updates=1200, lr=0.0003, gnorm=0.951, loss_scale=4, train_wall=252, wall=4669
2021-03-20 02:03:45 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 02:04:11 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 02:05:57 | INFO | train_inner | epoch 002:    365 / 941 loss=5.945, nll_loss=4.491, ppl=22.49, wps=62051, ups=0.39, wpb=157290, bsz=5609.5, num_updates=1300, lr=0.000325, gnorm=0.847, loss_scale=4, train_wall=252, wall=4923
2021-03-20 02:10:10 | INFO | train_inner | epoch 002:    465 / 941 loss=5.694, nll_loss=4.205, ppl=18.44, wps=62128.9, ups=0.39, wpb=157410, bsz=5533, num_updates=1400, lr=0.00035, gnorm=0.735, loss_scale=8, train_wall=252, wall=5176
2021-03-20 02:14:23 | INFO | train_inner | epoch 002:    565 / 941 loss=5.497, nll_loss=3.983, ppl=15.82, wps=62181.6, ups=0.4, wpb=157274, bsz=5464.6, num_updates=1500, lr=0.000375, gnorm=0.692, loss_scale=8, train_wall=251, wall=5429
2021-03-20 02:18:36 | INFO | train_inner | epoch 002:    665 / 941 loss=5.282, nll_loss=3.743, ppl=13.39, wps=62243.2, ups=0.4, wpb=157301, bsz=5557.7, num_updates=1600, lr=0.0004, gnorm=0.626, loss_scale=8, train_wall=251, wall=5682
2021-03-20 02:18:46 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 02:19:12 | INFO | fairseq.data.iterators | Data loading buffer is empty or nearly empty. This may indicate a data loading bottleneck, and increasing the number of workers (--num-workers) may help.
2021-03-20 02:21:59 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 4.0
2021-03-20 02:22:52 | INFO | train_inner | epoch 002:    766 / 941 loss=5.164, nll_loss=3.613, ppl=12.23, wps=61432.5, ups=0.39, wpb=157229, bsz=5504.3, num_updates=1700, lr=0.000425, gnorm=0.564, loss_scale=7, train_wall=254, wall=5938
2021-03-20 02:27:05 | INFO | train_inner | epoch 002:    866 / 941 loss=5.04, nll_loss=3.476, ppl=11.12, wps=61892.6, ups=0.39, wpb=156998, bsz=5501.7, num_updates=1800, lr=0.00045, gnorm=0.506, loss_scale=4, train_wall=252, wall=6191
2021-03-20 02:30:14 | INFO | fairseq_cli.train | begin validation on "valid" subset
2021-03-20 02:31:13 | INFO | valid | epoch 002 | valid on 'valid' subset | loss 4.841 | nll_loss 3.067 | ppl 8.38 | wps 149755 | wpb 8996.7 | bsz 315.6 | num_updates 1875 | best_loss 4.841
2021-03-20 02:31:13 | INFO | fairseq_cli.train | begin save checkpoint
2021-03-20 02:31:27 | INFO | fairseq.checkpoint_utils | saved checkpoint /apdcephfs/share_47076/qiqiliang/pruned/train_ende_BT/models_deen_baseline/checkpoint_best.pt (epoch 2 @ 1875 updates, score 4.841) (writing took 14.44751919992268 seconds)
2021-03-20 02:31:27 | INFO | fairseq_cli.train | end of epoch 2 (average epoch stats below)
2021-03-20 02:31:27 | INFO | train | epoch 002 | loss 5.725 | nll_loss 4.248 | ppl 19 | wps 45751.7 | ups 0.29 | wpb 157170 | bsz 5513 | num_updates 1875 | lr 0.00046875 | gnorm 0.752 | loss_scale 6 | train_wall 2368 | wall 6453
2021-03-20 02:31:27 | INFO | fairseq_cli.train | begin training epoch 2

**Comment by Sam131112** [comment]
Did you manage to reproduce the results ?  I also tried the setting and obtained following score 

BLEU+case.mixed+lang.de-en+numrefs.1+smooth.exp+test.wmt17+tok.13a+version.1.5.1 = 34.2 66.3/41.1/27.8/19.2 (BP = 0.983 ratio = 0.984 hyp_len = 63339 ref_len = 64399)
